numpy/core/tests/test_multiarray.py

```diff
@@ -6405,20 +6405,24 @@ def test_3d_tensor(self):
 
 class TestAlen(object):
     def test_basic(self):
-        m = np.array([1, 2, 3])
-        assert_equal(np.alen(m), 3)
+        with warnings.catch_warnings():
+            warnings.simplefilter("once")
```

Comment:
Can you just put the whole test into a with `pytest.warns(DeprecationWarning)`, that should work as well. I do not like the use of `"one"` filter, and why not just make sure that a warning is given (plus, the other version hides if there is a different type of warning)

Reply:
We have a file specifically for deprecations, `test_deprecations.py` - the test should go there so that it can use `assert_deprecated`

Reply:
There is a short test in `test_deprecations.py`, these are already existing tests elsewhere. Although we could move the, after the deprecation they will be deleted anyway.

Reply:
My mistake

Reply:
Done! I wasn't aware of the existence of `purest.warns(DeprecationWarning)`